File: hw/villagomez/u2/hw18ZooJS/ec.edu.espe.zoo.model/Lion.js
const Feline = require('./Feline'); 
const Cage = require('./Cage'); 


class Lion extends Feline {

    constructor(mane, id, description, bornOnDate, gender, numberOfBones, cage){
        super(id, description, bornOnDate, gender, numberOfBones, cage);
        this._mane = mane;
    }

    toString() {
        return `Lion{mane=${this._mane}, ${super.toString()}}`;
    }

    feed(food) {
        console.log(`Feeding a Lion with ${food} rats`);
    }

    feedQuantity(quantity, food) {
        console.log(`Feeding the lion with ${quantity} ${food}(s)`);
        console.log(`Number of legs ${this.numberOfLegs}`);
    }

    get mane() {
        return this._mane;
    }

    set mane(mane) {
        this._mane = mane;
    }

}

module.exports = Lion;

File: ws/gualotuna/u2/WS24Zoo/Zoo/src/main/java/ec/edu/espe/zoo/model/Lion.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.zoo.model;

import java.util.Date;

/**
 *
 * @author Brayan Gualotuña, Dev Dynasty, DCCO-ESPE
 */
public class Lion extends Feline{
    
    private boolean mane;

    @Override
    public String toString() {
        return "Lion{" + "mane=" + mane + super.toString() + '}';
    }

    public Lion(boolean mane, int id, String description, Date bornOnDate, char gender, Cage cage, int numberOfBones, int numberOfLegs) {
        super(id, description, bornOnDate, gender, cage, numberOfBones, numberOfLegs);
        this.mane = mane;
    }

    
    
    
    
    @Override
    public void feed(int food) {
        System.out.println("feeding a Lion");
    }

   
    public void feed(int quantity, Food food){
        System.out.println("feeding the lion with " + quantity + " " + food + "s");
        System.out.println("number of legs " + getNumberOfLegs());
    }
    
    
    /**
     * @return the mane
     */
    public boolean isMane() {
        return mane;
    }

    /**
     * @param mane the mane to set
     */
    public void setMane(boolean mane) {
        this.mane = mane;
    }
    
}
